### PR TITLE
#192/add item link on success

### LINKED ISSUE #192: Add a link to a newly created item on add-item/success page
# Expected Behaviour

```
As a user I expect to see a link to my newly created item after its successful creation 
```

# Current Behaviour
```
User is redirected to a page http://localhost:3000/add-item/success 
The message is shown: "You've successfully add an item"
Only "Add another item" button is shown 
```

## Steps to Reproduce
```
1. Run the application locally 
2. Log in as a user 
3. Go to add item page http://localhost:3000/add-item 
4. Fill in item name - test, description - test, postcode - CF1, condition - Good, category - Toys, check - Willing to post
5. Pick an image to upload (a valid jpeg file) 
6. Click  "add your item" button 
```
Screenshot 2024-04-29 at 00 57 38

### PR DESCRIPTION
**Pull request by Zu18**
# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review on my code
- [ ] I have made corresponding changes to the documentation (if any were needed)
- [x] My changes generate no new warnings or errors
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and previously existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

# Description
Closes #192  

feat: Implement URL parameters for success page and new item navigation

- Created a dynamic route for the success page under `app/(dashboard)/add-item/success/[itemId]/page.tsx`
- Updated the `SuccessPage` component to retrieve `itemId` of added item from URL parameters
- Modified `AddItemPage` to redirect to the success page using dynamic route with `itemId`
- Updated `insertRow` function to include returning data after insertion
- Improved URL clarity and SEO by using clean, hierarchical paths for navigation

style: Add container for buttons and set equal width

- Added a container around buttons on the SuccessPage for better layout structure
- Updated ButtonRounded component to accept a 'width' prop, allowing buttons to have equal width
- Applied the 'width' prop to buttons in the SuccessPage for consistent button size

Files changed:

- `app/(dashboard)/add-item/success/[id]/page.tsx`: Created a dynamic route for the success page.
- `SuccessPage.tsx`: Updated to retrieve `itemId` from URL parameters and added a container around the buttons.
- `AddItemPage.tsx`: Modified to redirect to the success page using dynamic route with `itemId`.
- `ButtonRounded.tsx`: Updated to accept a 'width' prop and applied it to the button's class for consistent sizing.
- `insertRow.ts`: Updated to return data after inserting a row.

### UI changes

- Buttons on the SuccessPage now have an equal width.
Screenshot from 2024-07-15 19-41-52

### CODE REVIEW

File: components/buttons/ButtonRounded.tsx [hunk 1]
```diff
@@ -6,17 +6,19 @@ type ButtonRoundedPropTypes = {
...
   children,
   clickHandler,
   isDisabled,
   type,
+  width,
 }) => {
   return (
     <button
-      className='button button-rounded disabled:bg-primaryGray'
+      className={`button button-rounded disabled:bg-primaryGray ${width || ''}`}
```
Comment (camelPhonso) [inline components/buttons/ButtonRounded.tsx:21]: I don't think we need the `||` here 🤔  it's just cleaner if we don't add an extra evaluation

**Review by camelPhonso — CHANGES_REQUESTED** [review]
This is a really neat solution Zukhra ✨ 

AND a beautiful PR description 😬 

I've asked for a few simple refactors but everything works and your changes were really neat :)

File: app/(dashboard)/add-item/page.tsx [hunk 2]
```diff
@@ -54,9 +54,15 @@ const AddItemPage = () => {
       ...data,
     };
 
-    await insertRow('items', dataItem);
-    reset();
-    router.push('/add-item/success');
+    const insertedItem = await insertRow('items', dataItem);
+    if (insertedItem && insertedItem.length > 0) {
+      const itemId = insertedItem[0].id;
+      console.log(itemId, 'add-item page');
```
[the remark this answers is not in the document]
Comment (Zu18) [reply app/(dashboard)/add-item/page.tsx]: Done

File: app/(dashboard)/add-item/page.tsx [hunk 3]
```diff
@@ -54,9 +54,15 @@ const AddItemPage = () => {
...
+      reset();
+      router.push(`/add-item/success/${itemId}`);
+    } else {
+      throw new Error('Failed to add the new item');
```
[the remark this answers is not in the document]
Comment (Zu18) [reply app/(dashboard)/add-item/page.tsx]: - Added try-catch block to onSubmit function to handle errors during the item addition process.
- Implemented guard clauses (!addedItem || addedItem.length === 0) in onSubmit to immediately throw an error if addedItem is null, undefined, or an empty array.
- Ensured the onSubmit function properly handles error conditions before proceeding with further operations.

File: app/(dashboard)/add-item/page.tsx [hunk 4]
```diff
@@ -54,14 +54,16 @@ const AddItemPage = () => {
...
+      if (!addedItem || addedItem.length === 0) {
+        throw new Error('Failed to add the new item');
+      }
+      const itemId = addedItem[0].id;
       reset();
       router.push(`/add-item/success/${itemId}`);
-    } else {
-      throw new Error('Failed to add the new item');
+    } catch (error) {
+      console.error('Error adding item:', error);
```
Comment (camelPhonso) [inline app/(dashboard)/add-item/page.tsx:66]: One final change and we're ready - can you also throw the error here please?

This way the error will be thrown from your evaluation to the `catch` block and then effectively break the page for us - which sounds counterintuitive but we want it to be flagged if any `nullish` values are coming back from the database so we can spot any future restructuring needs

**Review by camelPhonso — CHANGES_REQUESTED** [review]
Thanks for the clean follow up. 

Just requested one final addition and then I think we're good to merge this :)

File: app/(dashboard)/add-item/page.tsx (hunk 4 above)
Comment (Zu18) [reply app/(dashboard)/add-item/page.tsx:66]: Done

**Review by camelPhonso — APPROVED** [review]
Nice one Zukhra! Congrats on your first merge 🎉

**Pull request merged**